src/Simplify_LT.cpp

```diff
@@ -389,18 +389,14 @@ Expr Simplify::visit(const LT *op, ExprInfo *bounds) {
               rewrite(x < (x/c1)*c1, false, c1 > 0) ||
 
               // Cancel a division
-              rewrite((x + c1)/c0 < (x + c2)/c0, false, c0 > 0 && c1 >= c2) ||
-              rewrite((x + c1)/c0 < (x + c2)/c0, true, c0 > 0 && c1 <= c2 - c0) ||
+              rewrite((x + c1)/c0 < (x + c2)/c0, x%c0 < c2 - c1, c0 > 0) ||
               // c1 == 0
-              rewrite(x/c0 < (x + c2)/c0, false, c0 > 0 && 0 >= c2) ||
-              rewrite(x/c0 < (x + c2)/c0, true, c0 > 0 && 0 <= c2 - c0) ||
+              rewrite(x/c0 < (x + c2)/c0, x%c0 < c2, c0 > 0) ||
               // c2 == 0
-              rewrite((x + c1)/c0 < x/c0, false, c0 > 0 && c1 >= 0) ||
-              rewrite((x + c1)/c0 < x/c0, true, c0 > 0 && c1 <= 0 - c0) ||
+              rewrite((x + c1)/c0 < x/c0, x%c0 < -c1, c0 > 0) ||
 
               // The addition on the right could be outside
-              rewrite((x + c1)/c0 < x/c0 + c2, false, c0 > 0 && c1 >= c2 * c0) ||
-              rewrite((x + c1)/c0 < x/c0 + c2, true, c0 > 0 && c1 <= c2 * c0 - c0) ||
+              rewrite((x + c1)/c0 < x/c0 + c2, x%c0 < c2 * c0 - c1, c0 > 0) ||
```

Comment:
Synthesized version of these four added an extra one and made some constants non-constant:
```
 rewrite(((x + c0)/c1) < ((x/c1) + c2), (x % c1) < fold((c2*c1) - c0), c1 > 0) ||
 rewrite((x/c0) < ((x + y)/c0), (x % c0) < y, c0 > 0) ||
 rewrite((x/c0) < ((y + x)/c0), (x % c0) < y, c0 > 0) ||
 rewrite(((x + c0)/c1) < (x/c1), (x % c1) < fold(0 - c0), c1 > 0) ||
 rewrite(((x + c0)/c1) < ((x + c2)/c1), (x % c1) < fold(c2 - c0), c1 > 0)

```

Reply:
(I think that assumes they're not zero, because that would have constant-folded)